swift/Todo/Controller/AppController.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  AppController.swift
+//  Todo
+//
+//  Created by Callum Birks on 15/02/2023.
+//  Copyright © 2023 Couchbase. All rights reserved.
+//
+
+import os
+import SwiftUI
+import CouchbaseLiteSwift
+
+// QE:
+let kQEFeaturesEnabled = true
+
+// Logout Method
+enum LogoutMethod {
+    case closeDatabase, deleteDatabase;
+}
+
+class AppController {
+    public static let logger = Logger()
+    
+    public static func startSession(_ username: String, _ password: String) throws -> Bool {
+        Session.shared.start(username, password)
+        
+        try DB.shared.open()
+        
+        var success: Bool = true
+        
+        DB.shared.startReplicator { change in           
+            if let err = change.status.error as NSError?,
+               err.code == 401 { // 401 Unauthorized
+                success = false
```

Comment:
This doesn't seem to do anything so may be just logging that the authentication is failed or show a simple info dialog.

Reply:
The startSession() will always return true unless there is an exception. The startReplicator() doesn't block to return replication error so basically there is no point to set the success value. Regarding the behavior, this is fine as the app will continue to work even though the replicator is failed (weird but it's ok, just can't sync).